Analysis/Tasks/PWGHF/HFCandidateCreatorCascade.cxx

```diff
@@ -0,0 +1,304 @@
+  Configurable<double> d_maxR{"d_maxR", 200., "reject PCA's above this radius"};
+  Configurable<double> d_maxDZIni{"d_maxDZIni", 4., "reject (if>0) PCA candidate if tracks DZ exceeds threshold"};
+  Configurable<double> d_minParamChange{"d_minParamChange", 1.e-3, "stop iterations if largest change of any X is smaller than this"};
+  Configurable<double> d_minRelChi2Change{"d_minRelChi2Change", 0.9, "stop iterations is chi2/chi2old > this"};
+  Configurable<bool> b_doValPlots{"b_doValPlots", true, "do validation plots"};
+
+  // for debugging
+#ifdef MY_DEBUG
+  Configurable<std::vector<int>> v_labelK0Spos{"v_labelK0Spos", {729, 2866, 4754, 5457, 6891, 7824, 9243, 9810}, "labels of K0S positive daughters, for debug"};
+  Configurable<std::vector<int>> v_labelK0Sneg{"v_labelK0Sneg", {730, 2867, 4755, 5458, 6892, 7825, 9244, 9811}, "labels of K0S negative daughters, for debug"};
+  Configurable<std::vector<int>> v_labelProton{"v_labelProton", {717, 2810, 4393, 5442, 6769, 7793, 9002, 9789}, "labels of protons, for debug"};
+#endif
+
+  OutputObj<TH1F> hmass2{TH1F("hmass2", "2-prong candidates;inv. mass (#pi K) (GeV/#it{c}^{2});entries", 500, 0., 5.)};
+  OutputObj<TH1F> hCovPVXX{TH1F("hCovPVXX", "2-prong candidates;XX element of cov. matrix of prim. vtx. position (cm^{2});entries", 100, 0., 1.e-4)};
+  OutputObj<TH1F> hCovSVXX{TH1F("hCovSVXX", "2-prong candidates;XX element of cov. matrix of sec. vtx. position (cm^{2});entries", 100, 0., 0.2)};
+
+  double massP = RecoDecay::getMassPDG(kProton);
+  double massK0s = RecoDecay::getMassPDG(kK0Short);
+  double massPi = RecoDecay::getMassPDG(kPiPlus);
+  double massLc = RecoDecay::getMassPDG(pdg::code::kLambdaCPlus);
+  double mass2K0sP{0.};
+
+  void process(aod::Collisions const& collisions,
+               aod::HfTrackIndexCasc const& rowsTrackIndexCasc,
+               MyTracks const& tracks,
+               aod::V0Datas const& V0s
+#ifdef MY_DEBUG
+               ,
+               aod::McParticles& mcParticles
+#endif
+  )
+  {
+    // 2-prong vertex fitter
+    o2::vertexing::DCAFitterN<2> df;
+    df.setBz(d_bZ);
+    df.setPropagateToPCA(b_propDCA);
+    df.setMaxR(d_maxR);
+    df.setMaxDZIni(d_maxDZIni);
+    df.setMinParamChange(d_minParamChange);
+    df.setMinRelChi2Change(d_minRelChi2Change);
+    df.setUseAbsDCA(true);
+
+    // loop over pairs of track indeces
+    for (const auto& casc : rowsTrackIndexCasc) {
+      const auto& bach = casc.index0_as<MyTracks>();
+      auto trackParCovBach = getTrackParCov(bach);
+      const auto& v0 = casc.indexV0_as<o2::aod::V0Datas>();
+      const auto& posTrack = v0.posTrack_as<MyTracks>();
+      const auto& negTrack = v0.negTrack_as<MyTracks>();
+
+#ifdef MY_DEBUG
+      auto protonLabel = bach.mcParticleId();
+      auto labelPos = posTrack.mcParticleId();
+      auto labelNeg = negTrack.mcParticleId();
+      bool isLc = isLcK0SpFunc(protonLabel, labelPos, labelNeg, v_labelProton, v_labelK0Spos, v_labelK0Sneg);
```

Comment:
The names are confusing here.
* `protonLabel` should be `labelBach`
* These are not labels but indices.

Reply:
Why not "proton"? This is a proton. 
Can you tell me what the difference is? In the MC table, isn't the label the index?